### PR TITLE
SER-186 - resolve some Actions deprecation warnings

### PR DESCRIPTION
**Pull request by xaithe**

### CODE REVIEW

File: .github/workflows/api.yml [hunk 1]
```diff
@@ -25,17 +25,15 @@ jobs:
...
+
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-go@v2
-      - uses: actions/setup-python@v2
-      - name: setup go-testsum
-        uses: autero1/action-gotestsum@v1.0.0
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+      - uses: actions/setup-python@v4
```
Comment (mattmachell) [inline .github/workflows/api.yml:32]: Do we need python here at all as it's just a go workflow?

File: .github/workflows/behat.yml [hunk 2]
```diff
@@ -31,14 +31,14 @@ jobs:
...
       INFRA_VERSION: ${{ inputs.version_tag }}
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - uses: unfor19/install-aws-cli-action@v1
       - name: Install PHP 8.0
         uses: shivammathur/setup-php@v2
         with:
           php-version: "8.0.22"
       - name: Set up Python 3.8.1
```
Comment (mattmachell) [inline .github/workflows/behat.yml:40]: Worth updating the name here from 3.8.1

File: .github/workflows/behat.yml [hunk 3]
```diff
@@ -31,14 +31,14 @@ jobs:
...
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: "3.x"
```
Comment (mattmachell) [inline .github/workflows/behat.yml:43]: Might be worth pinning to a more specific version to avoid issues.

File: .github/workflows/web.yml [hunk 4]
```diff
@@ -5,11 +5,11 @@ jobs:
   test_web:
       runs-on: ubuntu-latest
       steps:
-        - uses: actions/checkout@v2
-        - name: Set up Python 3.8.1
-          uses: actions/setup-python@v2
+        - uses: actions/checkout@v3
+        - name: Set up Python 3.10.9
```
Comment (mattmachell) [inline .github/workflows/web.yml:9]: Not sure we need python here at all since everything in the build step is dockerised.

**Review by mattmachell — CHANGES_REQUESTED** [review]

**Review by charlesmmarshall — DISMISSED** [review]

**Review by charlesmmarshall — APPROVED** [review]

**Pull request merged**